### PR TITLE
Make sure events are handled completely before handling the next event

### PR DESCRIPTION
**Pull request by jonbretman**
Given the following component structure:

```
| component-a
| - component-b
| - - component-c
```

Assume `component-c` emits an event called `foo` and both `component-b` and `component-a` are listening for this event. When `component-b` receives this event it emits its own event called `bar` which `component-a` is listening to.

In the current implementation the following steps occur:

1. The `foo` event from `component-c` is received
2. The handler on `component-b` is called for the `foo` event
3. The `bar` event from `component-b` is received
4. The handler on `component-a` is called for the `bar` event
5. The `bar` event has no more listeners so baustein goes back to handling `foo`
6. The handler on `component-a` is called for the `foo` event

This is not ideal as in a complicated component structure baustein may be handling a 4th or even 5th generation event before finishing the handling of the first event.

This change would result in the following steps:

1. The `foo` event from `component-c` is received
2. The handler on `component-b` is called for the `foo` event
3. The `bar` event from `component-b` is received **and stored for processing later**
4. The handler on `component-a` is called for the `foo` event
5. The `foo` event has no more listeners so baustein **handles the next pending event**
6. The handler on `component-a` is called for the `bar` event

This change also provides support for calling `stopPropagation()` (see #7) on both DOM and custom events which if called will stop baustein from handing the event.

### CODE REVIEW

**Comment by jonbretman** [comment]
@djmc @iprignano

**Comment by iprignano** [comment]
Very cool, and thanks for the explanation :+1:

File: src/baustein.js [hunk 1]
```diff
@@ -366,6 +394,16 @@ export function isComponent(component) {
...
 
+    if (handlingEvent) {
+        pendingEvents.push(arguments);
+        return;
+    }
+
+    handlingEvent = true;
+
+    ensureEventHasMethod(event, 'stopPropagation', 'propagationStopped');
+    ensureEventHasMethod(event, 'preventDefault', 'defaultPrevented');
```
Comment (djmc) [inline src/baustein.js]: Maybe we should start looking at using real events? (https://developer.mozilla.org/en-US/docs/Web/API/Event/Event)

Or, is there a benefit to custom ones?

Comment (jonbretman) [inline src/baustein.js]: screen shot 2015-04-07 at 15 03 23

:cry: 

I suppose we could polyfil it for IE, but will probably be quite complicated.

Comment (djmc) [inline src/baustein.js]: Oh, I thought it was just `CustomEvent` that was like that. OK then...

**Comment by djmc** [comment]
:+1:

**Comment by Stereobit** [comment]
:+1:

**Pull request merged**